scripts/update-plugin-list.py

```diff
@@ -78,7 +78,7 @@ def iter_plugins():
         requires = "N/A"
         if info["requires_dist"]:
             for requirement in info["requires_dist"]:
-                if requirement == "pytest" or "pytest " in requirement:
+                if re.match(r"pytest(?![-.\w])", requirement):
```

Comment:
Just to be clear, the idea here is to match only `pytest`, but not `pytest-`, `pytest.` or `pytest `, correct?

Reply:
Pretty much. The idea is to match `pytest` as a package name. The package name ends either
- when the string ends, or
- when the next character cannot be part of the package name.

Package names can include any word character (`\w`, which includes digits and the underscore) as well as hyphens and periods ([PEP 508](https://peps.python.org/pep-0508/)). The big difference from `requirement.startswith("pytest ")` is that the character following the package name may also be a version specifier (`=`, `~`, `<`, or `>`). Indeed, the change in the plugin list after this MR will be precisely the addition of such requirement specifications.

Reply:
Great, thanks for the clarification! :+1: